src/snowflake/snowpark/_internal/temp_table_cleaner.py

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+import logging
+import threading
+from collections import defaultdict
+from queue import Queue
+from typing import TYPE_CHECKING, Dict
+
+if TYPE_CHECKING:
+    from snowflake.snowpark.session import Session  # pragma: no cover
+
+
+class TempTableCleaner:
```

Comment:
maybe just call this AutoManagedTableGarbageCollector, sounds fancier :)

Reply:
I'll avoid GarbageCollector because this is too specific to language runtime. How about just calling TempTableManager, which is more extensible for future usage?